tests/lib/c_lib/common/src/main.c

```diff
@@ -15,7 +15,7 @@
  * it guarantee that ALL functionality provided is working correctly.
  */
 
-#if defined(CONFIG_NEWLIB_LIBC) || defined(CONFIG_NATIVE_LIBC)
+#if defined(CONFIG_NATIVE_LIBC)
```

Comment:
Conditional declarations such as these could also be avoided if common libc includes were allowed with `CONFIG_NATIVE_LIBC`.